libraries/standard/mqtt/src/mqtt.c

```diff
@@ -938,7 +938,7 @@ static MQTTStatus_t handleIncomingAck( MQTTContext_t * pContext,
             break;
     }
 
-    if( ( status == MQTTSuccess ) && ( invokeAppCallback == true ) )
+    if( invokeAppCallback == true )
     {
         appCallback( pContext, pIncomingPacket, packetIdentifier, NULL );
```

Comment:
I think it would be helpful to update the callback signature to provide the de-serialization status for the application, as it may not be obvious to the customer about calling `MQTT_DeserializeAck` API to match the rejected subscription request with the packet ID when there are multiple "in-flight" requests. However, addition of a separate parameter would be a noise for all other packet types (PING, PUBLISH acks)....

Reply:
That's an option I considered, and weighed the drawback of adding another parameter to the callback with having the application call the deserializer. I thought this would be simpler, but figured it'd be discussed in this PR so a consensus could be reached. I can update this PR to update the callback if that's the direction we want to go

Reply:
Updated the callback